src/routers/alpha-router/alpha-router.ts

```diff
@@ -492,6 +504,16 @@ export class AlphaRouter
 
     this.swapRouterProvider =
       swapRouterProvider ?? new SwapRouterProvider(this.multicall2Provider);
+
+    if (tokenValidatorProvider) {
+      this.tokenValidatorProvider = tokenValidatorProvider;
+    } else if (this.chainId == ChainId.MAINNET) {
```

Comment:
Not currently as we don't have a V2 deployment on L2 for the token validator to use. It would be possible to just depend on a V2 fork on L2, but leaving that until we need it